map.go

```diff
@@ -579,6 +581,133 @@ func (m *Map) nextKey(key interface{}, nextKeyOut internal.Pointer) error {
 	return nil
 }
 
+// BatchLookup looks up many elements in a map at once
+// with the startKey being the first element to start
+// from.
+func (m *Map) BatchLookup(startKey, nextKey, keysOut, valuesOut interface{}, opts *BatchOptions) (int, error) {
```

Comment:
Can you document that:
* keysOut and valuesOut have to be slices, and may not be unsafe.Pointer
* both keysOut and valuesOut are always zeroed regardless of the count returned (because we can't take count into account in the call to unmarshalBytes)